app/src/main/java/com/example/taskyy/ui/viewmodels/ReminderViewModel.kt

```diff
@@ -171,51 +179,114 @@ class ReminderViewModel @Inject constructor(
-                    when (val save = agendaRepository.saveReminderToApi(newReminder)) {
+    private fun save(title: String, description: String, isDone: Boolean?) {
+        when (savedStateHandle.get<String>("agendaItem").toAgendaItemType()) {
+            AgendaItemType.REMINDER_ITEM -> {
+                val newReminder =
+                    Reminder(
+                        alarmType = _state.value.formattedReminderTime,
+                        title = title,
+                        description = description,
+                        timeInMillis = _dateTimeState.value.dateTime.toMillis(),
+                        id = UUID.randomUUID().toString(),
+                        agendaItem = AgendaItemType.REMINDER_ITEM,
+                        agendaAction = AgendaItemAction.CREATE
+                    )
+                viewModelScope.launch {
+                    when (val save = agendaRepository.saveReminderToDB(newReminder)) {
                         is Result.Success -> {
-                            eventChannel.send(ReminderEvent.SaveSuccessful)
+                            when (val save = agendaRepository.saveReminderToApi(newReminder)) {
+                                is Result.Success -> {
+                                    eventChannel.send(ReminderEvent.SaveSuccessful)
+                                }
+
+                                is Result.Error -> {
+                                    agendaRepository.addFailedReminderToRetry(newReminder)
+                                    eventChannel.send(ReminderEvent.SaveFailed(save.error.asUiText()))
+                                }
+                            }
                         }
+
                         is Result.Error -> {
-                            agendaRepository.addFailedReminderToRetry(newReminder)
                             eventChannel.send(ReminderEvent.SaveFailed(save.error.asUiText()))
                         }
                     }
                 }
-                is Result.Error -> {
-                    eventChannel.send(ReminderEvent.SaveFailed(save.error.asUiText()))
+            }
+
+            AgendaItemType.TASK_ITEM -> {
+                val newTask =
+                    Task(
+                        alarmType = _state.value.formattedReminderTime,
+                        title = title,
+                        isDone = isDone!!,
+                        description = description,
+                        timeInMillis = _dateTimeState.value.dateTime.toMillis(),
+                        id = UUID.randomUUID().toString(),
+                        agendaItem = AgendaItemType.REMINDER_ITEM,
+                        agendaAction = AgendaItemAction.CREATE,
+                    )
+                viewModelScope.launch {
+                    when (val save = agendaRepository.saveTaskToDB(newTask)) {
+                        is Result.Success -> {
+                            when (val save = agendaRepository.saveTaskToDB(newTask)) {
```

Comment:
Here, of course my previous feedback also applies that this logic goes in the repository. However, I don't understand why you're calling the save to DB function twice in a row. Did you mean to save it to the API instead?